pkg/resource/generator.go

```diff
@@ -103,6 +103,19 @@ func (g *Generator) syncStorage(cr *imageregistryv1.Config) error {
 			return err
 		}
 	}
+
+	// XXX
+	// This is a workaround for a bug that affected old releases. We may
+	// have multiple storage engines set on Status, that is not a valid
+	// situation. Here we check if this is the case and if yes we fix it
+	// by copying what we have on the Spec into the Status.
+	if _, err := storage.NewDriver(
+		&cr.Status.Storage, g.kubeconfig, g.listers,
+	); err == storage.ErrMultipleStorageConfigured {
```

Comment:
Nits:

1. Please reference the bug ID here - always good to refer to if this is an issue in the future.
2. /s/"old releases"/"OCP 4.2 and earlier"/

Reply:
Excellent points. Done.